src/ppx/grfx/grfx_swapchain.cpp

```diff
@@ -155,20 +202,46 @@ void Swapchain::DestroyColorImages()
 
 Result Swapchain::CreateDepthImages()
 {
-    if ((mCreateInfo.depthFormat != grfx::FORMAT_UNDEFINED) && mDepthImages.empty()) {
-        for (uint32_t i = 0; i < mCreateInfo.imageCount; ++i) {
-            grfx::ImageCreateInfo dpCreateInfo = ImageCreateInfo::DepthStencilTarget(mCreateInfo.width, mCreateInfo.height, mCreateInfo.depthFormat);
-            dpCreateInfo.ownership             = grfx::OWNERSHIP_RESTRICTED;
-            dpCreateInfo.DSVClearValue         = {1.0f, 0xFF};
+    if (!mDepthImages.empty()) {
+        return ppx::SUCCESS;
+    }
+    if (mCreateInfo.depthFormat == grfx::FORMAT_UNDEFINED) {
+        return ppx::SUCCESS;
+    }
+    for (uint32_t i = 0; i < mCreateInfo.imageCount; ++i) {
+        grfx::ImageCreateInfo dpCreateInfo = ImageCreateInfo::DepthStencilTarget(mCreateInfo.width, mCreateInfo.height, mCreateInfo.depthFormat);
+        dpCreateInfo.ownership             = grfx::OWNERSHIP_RESTRICTED;
+        dpCreateInfo.DSVClearValue         = {1.0f, 0xFF};
 
-            grfx::ImagePtr depthStencilTarget;
-            auto           ppxres = GetDevice()->CreateImage(&dpCreateInfo, &depthStencilTarget);
-            if (Failed(ppxres)) {
-                return ppxres;
-            }
+        grfx::ImagePtr depthStencilTarget;
+        auto           ppxres = GetDevice()->CreateImage(&dpCreateInfo, &depthStencilTarget);
+        if (Failed(ppxres)) {
+            return ppxres;
+        }
+
+        mDepthImages.push_back(depthStencilTarget);
+    }
+
+    return ppx::SUCCESS;
+}
 
-            mDepthImages.push_back(depthStencilTarget);
+Result Swapchain::CreateDepthImagesFromApiObjects(const std::vector<void*>& apiObjects)
+{
+    PPX_ASSERT_MSG(mDepthImages.empty(), "Swapchain depth images already initialized.");
```

Comment:
Feels slightly weird to error for that here, while above we return success with a no-op.